src/main/java/org/cascadebot/cascadebot/commands/management/permission/GroupPermissionListSubCommand.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2019 CascadeBot. All rights reserved.
+ * Licensed under the MIT license.
+ */
+
+package org.cascadebot.cascadebot.commands.management.permission;
+
+import java.util.ArrayList;
+import java.util.List;
+import net.dv8tion.jda.core.entities.Member;
+import org.cascadebot.cascadebot.commandmeta.CommandContext;
+import org.cascadebot.cascadebot.commandmeta.ICommandExecutable;
+import org.cascadebot.cascadebot.commandmeta.Module;
+import org.cascadebot.cascadebot.data.objects.GuildPermissions;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.permissions.objects.Group;
+import org.cascadebot.cascadebot.utils.pagination.Page;
+import org.cascadebot.cascadebot.utils.pagination.PageObjects;
+import org.cascadebot.cascadebot.utils.pagination.PageUtils;
+
+public class GroupPermissionListSubCommand implements ICommandExecutable {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if (context.getData().getPermissions().getGroups().isEmpty()) {
+            context.getTypedMessaging().replyWarning("You have no groups!");
+            return;
+        }
+
+        StringBuilder stringBuilder = new StringBuilder();
+        int i = 0;
```

Comment:
We could just use a traditional `for (int i = 0; i < context.getData().getPermissions().getGroups().size(); i++)` loop here

Reply:
I think it would perhaps be more readable